kratos/linear_solvers/amgcl_solver.h

```diff
@@ -568,6 +568,8 @@ class AMGCLSolver : public LinearSolver< TSparseSpaceType,
         else
             mBlockSize = ndof;
 
+        mBlockSize = rModelPart.GetCommunicator().GetDataCommunicator().SumAll(mBlockSize);
```

Comment:
no, this is not correct.

you should verify that the block size is the same om all processors. the only case which could be different is the zero case.

but you can not sum it

Reply:
This is outdated. I had a typo. It is now set to max. 

The problem is, I cannot check that everyone has the same block size, because if a partition does not have dof the size will be, by definition 0.

Notice that an access of a empty dofset was the cause of the crash in the first place...

Reply:
What we can do is, if the size has to be the same, is to broadcast the size of any non-empty process to here...